docs/rfcs/task.md

```diff
@@ -0,0 +1,761 @@
+significance of each particular requirement specified in this document.
+
+In brief: \"MUST\" (or \"REQUIRED\") means that the item is an absolute
+requirement of the specification; \"SHOULD\" (or \"RECOMMENDED\") means there
+may exist valid reasons for ignoring this item, but the full implications should
+be understood before doing so; and \"MAY\" (or \"OPTIONAL\") means that this
+item is optional, and may be omitted without careful consideration.
+
+
+## General Format
+
+The format is JSON, specifically a JSON object as a single line of text,
+terminated by a newline (U+000D).
+
+The JSON looks like this:
+
+    {"description":"One two three","status":"pending", ... }
+
+While this is not a valid task (there are missing fields), the format is
+illustrated.
+
+All attribute names are quoted with \" (U+0022). A name will always have a
+corresponding value, and if a value is blank, then the name/value pair is
+omitted from the line. Newline characters are not permitted within the value,
+meaning that a task consists of a single line of text.
+
+All data is UTF8.
+
+
+## Data Types
+
+There are five data types used in the task format.
+
+
+## Data Type: String
+
+Strings may consist of any UTF8 encoded characters.
+
+
+## Data Type: Fixed String
+
+A fixed string is one value from a set of acceptable values, such as a priority
+level, where the values may only be \"\", \"L\", \"M\" or \"H\".
+
+
+## Data Type: UUID
+
+A UUID is a 32-hex-character lower case string, formatted in this way:
+
+    xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+
+An example:
+
+    296d835e-8f85-4224-8f36-c612cad1b9f8
+
+
+## Data Type: Integer
+
+Integers are rendered in a simple fashion:
+
+    123
+
+
+## Data Type: Date
+
+Dates are rendered in ISO 8601 combined date and time in UTC format using the
+template:
+
+    YYYYMMDDTHHMMSSZ
+
+An example:
+
+    20120110T231200Z
+
+No other formats are supported.
+
+
+## Data Type: Duration
+
+Duration values represent a time period. They take the form:
+
+    [[<sign>] <number>] <unit>
+
+Some examples include:
+
+-   -3days
+-   annual
+-   4hrs
+
+The supported units are:
+
+-   annual
+-   biannual
+-   bimonthly
+-   biweekly
+-   biyearly
+-   daily
+-   days
+-   day
+-   d
+-   fortnight
+-   hours
+-   hour
+-   hrs
+-   hr
+-   h
+-   minutes
+-   mins
+-   min
+-   monthly
+-   months
+-   month
+-   mnths
+-   mths
+-   mth
+-   mos
+-   mo
+-   quarterly
+-   quarters
+-   qrtrs
+-   qtrs
+-   qtr
+-   q
+-   seconds
+-   secs
+-   sec
+-   s
+-   semiannual
+-   sennight
+-   weekdays
+-   weekly
+-   weeks
+-   week
+-   wks
+-   wk
+-   w
+-   yearly
+-   years
+-   year
+-   yrs
+-   yr
+-   y
+
+Note that some values lack precision, for example \"2q\" means two quarters, or
+half a year.
+
+Note that not all combinations of and make sense, for example \"3annual\" makes
+no sense, but evaluates to \"3years\".
+
+
+## The Attributes
+
+Here are the standard attributes that may comprise a task:
+
+<table>
```

Comment:
Reformat to Markdown table?

Reply:
I did a few of these, where the Markdown table could reasonably replicate the HTML.

But on some of the never-finished RFCs with complicated tables it just wasn't worth the effort, so I left the tables as-is.

Reply:
(in the end this was just the "plans" RFC, which is from 2016 and maybe should just be rm -rf'd?

Reply:
Deleting `plans.md` if it is outdated is fine with me. But as I am not directly involved in Taskwarrior development, I would delegate the final decision to @tbabej  😃